src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -153,16 +161,25 @@ public ProgramEntity getProgram(@NonNull String name, boolean allowInactive) {
   }
 
   public ProgramEntity createWithSideEffect(
-      @NonNull Program program, Consumer<ProgramEntity> consumer) {
-    val programEntity = createProgram(program);
+      @NonNull Program program, Consumer<ProgramEntity> consumer, UUID dataCenterId) {
+    val programEntity = createProgram(program, dataCenterId);
     consumer.accept(programEntity);
     return programEntity;
   }
 
-  public ProgramEntity createProgram(@NonNull Program program)
+  public ProgramEntity createProgram(@NonNull Program program, UUID dataCenterId)
       throws DataIntegrityViolationException {
     val programEntity = programConverter.programToProgramEntity(program);
 
+    if (dataCenterId != null) {
+      val dataCenterEntity = dataCenterRepository.findById(dataCenterId);
+      if (!dataCenterEntity.isEmpty()) {
+        programEntity.setDataCenterId(dataCenterId);
+      } else {
+        throw new BadRequestException("DataCenterId '" + dataCenterId + "' not found");
+      }
+    }

```

Comment:
Do we handle the case where `dataCenterId` is null? Should also be a bad request, ya?

Reply:
Handled the case where datacenterId is null